### PR TITLE
test: Cleanup GPU-AV BufferDeviceAddress test

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/4983

The old test was incorrect how it mapped the memory (commented below)

Incase the test still hangs on CI, it will be easier to diagnosis now that things are broken up

Also created a dedicated positive file

### CODE REVIEW

File: tests/unit/gpu_av_buffer_device_address.cpp [hunk 1]
```diff
@@ -166,102 +113,172 @@ TEST_F(NegativeGpuAVBufferDeviceAddress, DISABLED_Basic) {
...
-        m_commandBuffer->BeginRenderPass(m_renderPassBeginInfo);
-        vk::CmdBindPipeline(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_GRAPHICS, pipe.Handle());
-        vk::CmdBindDescriptorSets(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout.handle(), 0, 1,
-                                  &descriptor_set.set_, 0, nullptr);
-        vk::CmdDraw(m_commandBuffer->handle(), 3, 1, 0, 0);
-        m_commandBuffer->EndRenderPass();
-        m_commandBuffer->end();
-
-        for (const auto &test : testcases) {
-            auto *data = static_cast<VkDeviceAddress *>(buffer0.memory().map());
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_buffer_device_address.cpp:187]: the data is 

```
            layout(push_constant) uniform ufoo {
                bufStruct data;
                int nWrites;
            } u_info;
```

but we were pushing 16 bytes here

File: tests/unit/gpu_av_buffer_device_address.cpp [hunk 2]
```diff
@@ -102,58 +49,58 @@ TEST_F(NegativeGpuAVBufferDeviceAddress, DISABLED_Basic) {
...
-    }
-    // descriptor set version
-    {
-        OneOffDescriptorSet descriptor_set(m_device, {{0, VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1, VK_SHADER_STAGE_ALL, nullptr}});
-
-        const vkt::PipelineLayout pipeline_layout(*m_device, {&descriptor_set.layout_});
-        VkDescriptorBufferInfo buffer_test_buffer_info = {};
-        buffer_test_buffer_info.buffer = buffer0.handle();
-        buffer_test_buffer_info.offset = 0;
-        buffer_test_buffer_info.range = sizeof(uint32_t);
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_buffer_device_address.cpp:142]: we were only updating the first 4 bytes of the buffer, not all 12 bytes, so likely some machines were just luckily pushing the whole cache line, but others were not

**Review by arno-lunarg — APPROVED** [review]
LGTM, wondering about the crash ...

**Comment by spencer-lunarg** [comment]
ok, seems the `ReadAfterPointer` test is not be detecting on some machines and then crashing on one... so assume the issue is the shader instrumentation

**Review by jeremyg-lunarg — APPROVED** [review]
Thanks for figuring this out

**Pull request merged**